From cdae742e74e7bdc4a764007b1eee46630222f438 Mon Sep 17 00:00:00 2001
From: Peng Fan <peng.fan@nxp.com>
Date: Sat, 9 Dec 2017 18:44:37 +0800
Subject: [PATCH 3062/5242] MLK-17142 arm64: dts: 8qxp: introduce dom0 dts

commit  45ba18a5abf9188806eb8c0a2a7625a8e21e123a from
https://source.codeaurora.org/external/imx/linux-imx.git

Introudce dom0 dts.
disable rtc, we do not have sip handler in xen, also need to modify the
driver to use hvc call.
remove gpu ss reg property to avoid overriding dom0 mapping.
Modify lpuart0 interrupt-parent to use gic, because xen could only
handle gic.

Signed-off-by: Peng Fan <peng.fan@nxp.com>
Reviewed-by: Bai Ping <ping.bai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/Makefile             |    1 +
 .../boot/dts/freescale/fsl-imx8qxp-mek-dom0.dts    |   44 ++++++++++++++++++++
 2 files changed, 45 insertions(+)
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-dom0.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index f23c164..bb11784 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -40,6 +40,7 @@ dtb-$(CONFIG_ARCH_FSL_IMX8QM) += fsl-imx8qm-lpddr4-arm2.dtb \
 				 fsl-imx8qm-lpddr4-arm2-it6263-rm67191.dtb
 dtb-$(CONFIG_ARCH_FSL_IMX8QXP) += fsl-imx8qxp-lpddr4-arm2.dtb \
 				  fsl-imx8qxp-mek.dtb \
+				  fsl-imx8qxp-mek-dom0.dtb \
 				  fsl-imx8qxp-mek-enet2.dtb \
 				  fsl-imx8qxp-mek-lvds0-it6263.dtb \
 				  fsl-imx8qxp-mek-lvds1-it6263.dtb \
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-dom0.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-dom0.dts
new file mode 100644
index 0000000..88942ca
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-dom0.dts
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2017 NXP
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "fsl-imx8qxp-mek.dts"
+
+
+/ {
+	chosen {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		/delete-property/ bootargs;
+		/delete-property/ stdout-path;
+		module@0 {
+			bootargs = "earlycon=xen console=hvc0 clk_ignore_unused loglevel=8 root=/dev/mmcblk1p2 rootwait rw video=imxdpufb5:off video=imxdpufb6:off video=imxdpufb7:off";
+                       compatible = "xen,linux-zimage", "xen,multiboot-module";
+		       /* The size will be override by uboot command */
+                       reg = <0x00000000 0x80a00000 0x00000000 0xf93a00>;
+               };
+	};
+};
+
+&lpuart0 {
+	interrupt-parent = <&gic>;
+};
+
+&imx8_gpu_ss {
+	/delete-property/ reg;
+	/delete-property/ reg-names;
+};
+
+&rtc {
+	status = "disabled";
+};
-- 
1.7.9.5

